genCode.py

```diff
@@ -77,10 +77,24 @@ def createStaticClient(name, api, genAsync=False):
 
     lines.append('    classOptions = {')
 
-    copiedOptions = ('baseUrl', 'exchangePrefix')
+    # if serviceName is not set, get it from the baseUrl, exchangePrefix, or manifest name
+    # this can be removed when all services have serviceName
+    if 'serviceName' not in api:
+        if 'baseUrl' in api:
+            api['serviceName'] = api['baseUrl'].split('/')[2].split('.')[0]
```

Comment:
This could wouldn't ever get a bsaeUrl like that, because it's only looking at baseUrls in API definitions (not in constructor options), and only for old services where `serviceName` wasn't given (so, effectively, just aws-provisioner).  So we could just guess 'aws-provisioner' and actually be right every time :)

But as I said in my reply, we could fix aws-provisioner to provide serviceName, and then never need to apply this guessing at all.

Reply:
I think that a point release to lib-api for aws-provisioner is the way to go then.